tests/src/JIT/Regression/CLR-x86-JIT/V1.2-Beta1/b103058/b103058.cs

```diff
@@ -31,17 +32,43 @@ private static unsafe bool CheckDoubleAlignment1(VT* p)
         }
     }
 
+    private static unsafe bool CheckAlignment1(VT* p)
+    {
+        Console.WriteLine("Address {0}", (IntPtr)p);
+        if ((int)(long)p % 4 != 0)
+        {
+            Console.WriteLine("not 4-byte aligned");
+            return false;
+        }
+        else
+        {
+            return true;
+        }
+    }
+
     public static int Main()
     {
         VT vt1 = new VT();
         VT vt2 = new VT();
         bool retVal;
 
-        retVal = CheckDoubleAlignment1(&vt1);
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || (RuntimeInformation.ProcessArchitecture != Architecture.X86))
```

Comment:
It would be better to have this check inside `CheckDoubleAlignment1` method to have less code duplication.